code/__DEFINES/machines.dm

```diff
@@ -106,3 +106,8 @@
 //cloning defines. These are flags.
 #define CLONING_SUCCESS (1<<0)
 #define CLONING_DELETE_RECORD (1<<1)
+
+//these are used to tell the DNA modifier if a plant gene cannot be extracted or modified.
+#define PLANT_GENE_IMMUTABLE	0
+#define PLANT_GENE_REMOVABLE	1
+#define PLANT_GENE_EXTRACTABLE	2

```

Comment:
These can probably benefit from being flags.

Reply:
What is the advantage over using defines?

Reply:
I meant that you could put flags on the genes that tell you, this can't be removed or this can't be extracted.

The benefit of flag is that you can apply all of them at once. For example it would be way easier to make a gene that can't be extracted or removed by just doing ``gene_flags = GENE_NOEXTRACT|GENE_NOREMOVE``

The benefit of doing that is that is easier to expand upon.

Reply:
what is an immutable gene?

Removable/Extractable overlap so those wouldn't be good for flags since to extract a gene you need to remove it from the seed, ruining it in the process.

I guess you could just refluff it if that's the only issue but 🤷‍♂